### PR TITLE
unmatch docs after deleting

### PR DESCRIPTION
**Pull request by Serhiy91**

### CODE REVIEW

**Comment by Dartv** [comment]
Where are tests?

**Comment by Serhiy91** [comment]
@Dartv added

File: share/services/__tests__/customer-segment-service.test.js [hunk 1]
```diff
@@ -40,4 +43,22 @@ describe('Customer segment service', () => {
...
+      },
+    };
+    const valuePropositionId = await ValuePropositions.insert(valuePropositionData);
+
+    await CustomerSegmentService.delete({ _id: customerSegmentId }, context);
+
+    const customerSegment = await CustomerSegments.findOne({ _id: customerSegmentId });
+    const valueProposition = await ValuePropositions.findOne({ _id: valuePropositionId });
+    expect(customerSegment).toBe(null);
+    expect(valueProposition).toMatchObject({ _id: valuePropositionId });
```
Comment (Dartv) [inline share/services/__tests__/customer-segment-service.test.js]: `toMatchObject` skips other props so it doesn't ensure the `matchedTo` is removed. Probably should use `toEqual` there.

Comment (Dartv) [reply share/services/__tests__/customer-segment-service.test.js]: This should also test if `unmatch` method is actually called and called with correct arguments. So mock it like this maybe:
` CustomerSegmentService.unmatch = jest.fn(CustomerSegmentUnmatch)`

**Pull request merged**